sdk/core/azure-core/tests/test_rest_http_request.py

```diff
@@ -569,10 +570,35 @@ def test_multipart_tuple_input_multiple_with_filename_and_content_type(filebytes
         files=[("file", ("first file", filebytes, "image/pdf")), ("file2", ("second file", filebytes, "image/png"))],
     )
 
-    assert request.content == {
-        "file": ("first file", filebytes, "image/pdf"),
-        "file2": ("second file", filebytes, "image/png"),
-    }
+    assert request.content == [
+        ("file", ("first file", filebytes, "image/pdf")),
+        ("file2", ("second file", filebytes, "image/png")),
+    ]
+
+
+def test_multipart_tuple_input_multiple_same_name(client):
+    request = HttpRequest(
+        "POST",
+        url="/multipart/tuple-input-multiple-same-name",
+        files=[
+            ("file", ("firstFileName", NamedIo("firstFile"), "image/pdf")),
+            ("file", ("secondFileName", NamedIo("secondFile"), "image/png")),
+        ],
+    )
+    client.send_request(request).raise_for_status()
+
+
+def test_data_and_file_input_same_name(client):
+    request = HttpRequest(
+        "POST",
+        url="/multipart/data-and-file-input-same-name",
+        data={"message": "Hello, world!"},
```

Comment:
This will set the field "message" with the string hello world, is that really what we expect?

```
--f803752c040f939e434591439d40550b
Content-Disposition: form-data; name="message"

Hello, world!
--f803752c040f939e434591439d40550b
Content-Disposition: form-data; name="file"; filename="firstFileName"
Content-Type: image/pdf


--f803752c040f939e434591439d40550b--
Content-Disposition: form-data; name="file"; filename="secondFileName"
Content-Type: image/png


--f803752c040f939e434591439d40550b--
```

Reply:
yep, talked offline with @lmazuel we are good with this test